src/CommandingOfficers/CommanderAve.java

```diff
@@ -423,72 +481,38 @@ protected void perform(MapMaster gameMap)
-        tilesInRange.addAll(Utils.findLocationsInRange(gameMap, prop, CommanderAve.MAX_SNOW_SPREAD_RANGE+1, GLACIO_MAX_SNOW_SPREAD_RANGE));
-      }
-      for( XYCoord tile : tilesInRange )
-      {
-        Ave.snowMap[tile.xCoord][tile.yCoord] = CommanderAve.SNOW_THRESHOLD;
-        if( gameMap.getEnvironment(tile).weatherType != Weathers.SNOW )
+        if( Ave.snowMap[coord.xCoord][coord.yCoord] < CommanderAve.SNOW_THRESHOLD )
         {
-          tileChanges.add(new MapChangeEvent.EnvironmentAssignment(tile, Environment.getTile(gameMap.getEnvironment(tile).terrainType, Weathers.SNOW), 1));
+          Ave.snowMap[coord.xCoord][coord.yCoord] = CommanderAve.SNOW_THRESHOLD;
+          if( gameMap.getEnvironment(coord).weatherType != Weathers.SNOW )
+          {
+            tileChanges.add(new MapChangeEvent.EnvironmentAssignment(coord, Environment.getTile(gameMap.getEnvironment(coord).terrainType, Weathers.SNOW), 1));
+          }
         }
       }
 
-      // Change terrain to snow around each of Ave's units, and damage trees and enemies.
-      HashSet<XYCoord> tilesSeen = new HashSet<XYCoord>();
-      for( Unit unit : myCommander.units )
+      // Freeze enemies around each of Ave's units or buildings.
+      tilesInRange = Utils.findLocationsNearUnits(gameMap, Ave, GLACIO_FREEZE_RANGE);
+      tilesInRange.addAll(Utils.findLocationsNearProperties(gameMap, Ave, GLACIO_FREEZE_RANGE));
+      for( XYCoord coord : tilesInRange )
       {
-        // Two-tile radius of effect around each unit.
-        ArrayList<XYCoord> tiles = Utils.findLocationsInRange(gameMap, new XYCoord(unit.x, unit.y), 0, 2);
-        for( XYCoord coord : tiles )
+        // Freeze each nearby enemy.
+        Location loc = gameMap.getLocation(coord);
+        if( null != loc.getResident() && myCommander.isEnemy(loc.getResident().CO) )
         {
-          // Don't Glacio the same tile twice.
-          if( tilesSeen.contains(coord) ) continue;
-          tilesSeen.add(coord);
-
-          // Add snow around friendlies.
-          Location loc = gameMap.getLocation(coord);
-
-          // Turn the spaces around each unit to snow.
-          Environment tileEnvi = loc.getEnvironment();
-          if((tileEnvi.weatherType != Weathers.SNOW) || (tileEnvi.terrainType == TerrainType.FOREST))
-          {
-            // Destroy any forests. Big hail, man.
-            TerrainType newTerrain = (loc.getEnvironment().terrainType == TerrainType.FOREST)
-                ? TerrainType.GRASS : loc.getEnvironment().terrainType;
-            tileChanges.add(new MapChangeEvent.EnvironmentAssignment(coord, Environment.getTile(newTerrain, Weathers.SNOW), 1));
-          }
-
-          // Damage each enemy nearby.
-          if( null != loc.getResident() && myCommander.isEnemy(loc.getResident().CO) )
-          {
-            loc.getResident().alterHP(-2);
-          }
+          // TODO: Apply Freezing.
```

Comment:
Should frozen units be able to counterattack?
Should they take extra damage?
If you care at all, this is the diff for when I added stunning for VB https://github.com/ThislsAUsername/DefendPeace/commit/a6ae33b9fd7858694154bc07f1327151373faf5d

Reply:
I was kind of assuming stunned == not able to counterattack, but no extra damage taken. What does VB do on the carts? It seems sensible to just use one mechanism for stunning, whatever the cause.

Reply:
Units VB has stunned can shoot back on the carts. I know it's silly, but it's definitely a thing.
Also worth noting is that Eagle can make stunned units act out of stun.

The nomenclature you used was "freeze", so I assumed it might have some different mechanics.
Disabling counterattacks is something you'd kind of assume any sort of stun would do.
Boosting damage taken is something freezing status frequently does in games.

Also, both of the above points are use cases for third-party combat modifiers. You don't want to do a UnitRemodel on any other CO's units, so fiddling with defense is right out.
Buffing attacks won't work, since that's a global effect.
Disabling counterattack or boosting damage (unless it becomes a feature of Unit, which seems silly) has to be a combat modifier, and Ave can't do it (atm) if someone else attacks the enemy she froze.

Reply:
> both of the above points are use cases for third-party combat modifiers

Disabling counterattacks wouldn't require third-party interference - the combat engine would just have to know what stun means. Same would go for increasing damage, potentially, but I wasn't looking to make that happen.

> Disabling counterattacks is something you'd kind of assume any sort of stun would do.

Yes, but it also feels pretty strong, which is probably why they allowed stunned counterattacking on the carts. Any preference which way we go? I'd like to stick with one "stun" effect unless there's an awfully good reason not to. Should we match VB for tradition's sake, or go our own way and do the more intuitive/OP thing?

Reply:
> > both of the above points are use cases for third-party combat modifiers
> 
> Disabling counterattacks wouldn't require third-party interference - the combat engine would just have to know what stun means. Same would go for increasing damage, potentially, but I wasn't looking to make that happen.

Perhaps stun should natively negate counterattacks (though, as you say, 'tis quite strong), but I wouldn't want to add "target takes increased damage" as a rider to "stun", especially since it should be of configurable magnitude.
I was also thinking the counterattack negation might be a special thing for Ave, seeing as she supposedly freezes the enemies/coats 'em in ice, not just puts 'em out of sorts.
Perhaps it would satisfy everyone to just have a stat delta per unit? That seems like it could get out of whack pretty easily, if someone forgets or applies twice or what-have-you.

> > Disabling counterattacks is something you'd kind of assume any sort of stun would do.
> 
> Yes, but it also feels pretty strong, which is probably why they allowed stunned counterattacking on the carts. Any preference which way we go? I'd like to stick with one "stun" effect unless there's an awfully good reason not to. Should we match VB for tradition's sake, or go our own way and do the more intuitive/OP thing?

Looks like you've solved this already.

Reply:
Yeah, I guess. For icons though, there isn't really one that matches both "freeze from cold" and "stun from lightning". Almost seems worth differentiating just for that, except then we'd come up with more causes of stunning.